poetry/core/vcs/git.py

```diff
@@ -14,7 +14,8 @@
     "port": r"\d+",
     "path": r"[\w~.\-/\\]+",
     "name": r"[\w~.\-]+",
-    "rev": r"[^@#]+",
+    "rev": r"[^@#?]+",
+    "subdir": r"[\w\-/\\]+",
```

Comment:
Linking to previous discussion concerning this bit. https://github.com/python-poetry/poetry/pull/1822#discussion_r399669157